api/controllers/shabads.js

```diff
@@ -258,6 +258,65 @@ exports.search = async (req, res) => {
   }
 };
 
+exports.resultsInfo = async (req, res) => {
+  const { VerseID } = req.params;
+  const results = 20;
+  let page = parseInt(req.query.page, 10) || 1;
+  if (lib.isListOfNumbers(VerseID)) {
+    let conn;
+    try {
+      conn = await req.app.locals.pool.getConnection();
+
+      const q = `SELECT ${allColumns} ${allFrom}
+           WHERE v.ID IN (${VerseID})`;
+
+      const row = await conn.query(`SELECT COUNT(*) FROM (${q}) AS count`, []);
+
+      const totalResults = row[0]['COUNT(*)'];
+      const totalPages = Math.ceil(totalResults / results);
+      if (page > totalPages) {
+        page = totalPages;
+      }
+      const resultsInfo = {
+        totalResults,
+        pageResults: totalResults,
+        pages: {
+          page,
+          resultsPerPage: results,
+          totalPages,
+        },
+      };
+
+      if (totalResults > 0) {
+        if (page < totalPages) {
+          req.query.page = page + 1;
+          resultsInfo.pages.nextPage = `${req.protocol}://${req.get('host')}${req.baseUrl}${
+            req.path
+          }?${Object.keys(req.query)
+            .map(key => `${key}=${encodeURIComponent(req.query[key])}`)
+            .join('&')}`;
```

Comment:
Maybe we can divide this into two different statements (for base path and query keys.

Also, can we use a URL builder instead?

Something like:

```js
const url = require('url');
const myURL = url.format({
  protocol: req.protocol,
  hostname: req.get('host'),
  pathname: '/some/path',
  query: {
    key: 'value'
  }
});
```

Reply:
This is a common pattern used through out the routes, created a separate issue to change it everywhere.
https://github.com/KhalisFoundation/banidb-api/issues/270